opentracing-okhttp3/src/main/java/io/opentracing/contrib/okhttp3/TracingCallFactory.java

```diff
@@ -0,0 +1,124 @@
+                    .startActive();
+
+            /**
+             * In case of exception network interceptor is not called
+             */
+            OkHttpClient.Builder okBuilder = okHttpClient.newBuilder();
+            okBuilder.networkInterceptors().add(0, new NetworkInterceptor(tracer, activeSpan.context(), decorators));
+
+            final ActiveSpan.Continuation continuation = activeSpan.capture();
+            okBuilder.interceptors().add(0, new Interceptor() {
+                @Override
+                public Response intercept(Chain chain) throws IOException {
+                    ActiveSpan activate = continuation.activate();
+                    try {
+                        return chain.proceed(chain.request());
+                    } catch (Exception ex) {
+                        for (OkHttpClientSpanDecorator spanDecorator : decorators) {
+                            spanDecorator.onError(ex, activate);
+                        }
+                        throw ex;
+                    } finally {
+                        activate.deactivate();
+                    }
+                }
+            });
+            return okBuilder.build().newCall(request);
+        } catch (Throwable ex) {
+            for (OkHttpClientSpanDecorator spanDecorator: decorators) {
+                spanDecorator.onError(ex, activeSpan);
+            }
+            throw ex;
+        } finally {
+            activeSpan.deactivate();
+        }
+    }
+
+    static class NetworkInterceptor implements Interceptor {
+        public SpanContext parentContext;
+        public Tracer tracer;
+        public List<OkHttpClientSpanDecorator> decorators;
+
+        NetworkInterceptor(Tracer tracer, SpanContext spanContext, List<OkHttpClientSpanDecorator> decorators) {
+            this.parentContext = spanContext;
+            this.tracer = tracer;
+            this.decorators = decorators;
+        }
+
+        @Override
+        public Response intercept(Chain chain) throws IOException {
+            try (ActiveSpan networkSpan = tracer.buildSpan(chain.request().method())
+                        .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
+                        .asChildOf(parentContext)
+                        .startActive()) {
+
+                for (OkHttpClientSpanDecorator spanDecorator: decorators) {
+                    spanDecorator.onRequest(chain.request(), networkSpan);
+                }
+
+                Request.Builder requestBuilder = chain.request().newBuilder();
+                tracer.inject(networkSpan.context(), Format.Builtin.HTTP_HEADERS, new RequestBuilderInjectAdapter(requestBuilder));
+                Response response = chain.proceed(requestBuilder.build());
+
+                for (OkHttpClientSpanDecorator spanDecorator: decorators) {
+                    spanDecorator.onResponse(chain.connection(), response, networkSpan);
+                }
+
```

Comment:
Would it not be better to handle the exception (and onError decoration) here, so all decoration handled in the same interceptor?